content/blog/2019-10-23_react-typescript-starter.md

```diff
@@ -0,0 +1,464 @@
+
+I am also going to use [yarn](https://yarnpkg.com/lang/en/) as package manager but `npm` will work as well.
+
+You can view the end result in <https://github.com/Keats/webpack-react-typescript>.
+
+## TypeScript setup
+[TypeScript](https://www.typescriptlang.org/) is the most important tool for me in the frontend world. I first talked about 
+it on this blog [in 2014](@./blog/2014-5-4_switching-to-typescript.md) and it only got better with time. This is a superset of JavaScript, simply adding 
+types to it. I don't think static typing in dynamic languages are controversial anymore, seeing how [many](https://sorbet.org/) [languages](http://mypy-lang.org/) adopt it.
+Static typing is the ultimate form of documentation as it HAS to be correct.
+
+Let's add it to the project:
+
+```bash
+$ yarn add typescript --dev
+```
+
+This will create your project `package.json` as well as a lock file: `yarn.lock`.
+
+The next step is to create a `tsconfig.json`, containing our TypeScript configuration:
+
+```json
+{
+  "compilerOptions": {
+    "module": "es6",
+    "moduleResolution": "node",
+    "esModuleInterop": true,
+    "target": "es6",
+    "jsx": "react",
+    "sourceMap": true,
+    "strict": true,
+    "forceConsistentCasingInFileNames": true,
+    "noUnusedLocals": true,
+    "noUnusedParameters": true,
+    "noImplicitReturns": true,
+    "noFallthroughCasesInSwitch": true,
+    "allowUnreachableCode": false
+  },
+  "include": [
+    "./src/**/*"
+  ]
+}
+```
+
+Let's go through them quickly:
+
+- `module` and `esModuleInterop`: ensures we use the latest module system and compatible with the Babel ecosystem
+- `moduleResolution`: look into the `node_modules` folder for packages
+- `target`: we target modern browsers so ES6 is fine here. If you need IE11 support, change it to `es5`
+- `jsx`: we use React and transpile from TypeScript directly so we use `react`
+- `sourceMap`: we want them
+- `strict`: enable all the strict flags automatically
+
+All the other options are adding some additional checks, such as not allowing unused variables and a list 
+of all the options can be found in [TypeScript documentation](https://www.typescriptlang.org/docs/handbook/compiler-options.html).
+Lastly, we set up an include path containing `"./src/**/*"`: we will put all our code in a `src` folder.
+
+In this setup, we will only use TypeScript and not [Babel](https://babeljs.io/). Is it possible to use TypeScript for
+typechecking only and pass the code to Babel for transpilation but, unless you need a specific Babel plugin, there are no 
+reasons to do that nowadays. You will have to wait a few more months before getting stage 3 features - not a big deal.
+
+The commit for that step can be seen in <https://github.com/Keats/webpack-react-typescript/commit/a083ce1511b15c51d4a8d3ef95f2da05a669758b>.
+
+## Webpack setup
+Let's start the slightly more complex part! Once you understand how Webpack works, this is actually pretty simple.
```

Comment:
> Once you understand how Webpack works
:trollface: 

Might be worth adding that webpack has a fair learning curve attached.

Reply:
Je sais pas si c'est encore vrai avec webpack 4, c'est bcp plus simple qu'avant